### PR TITLE
Add Cups choice options

### PR DESCRIPTION
**Pull request by esutley**
Makes Cups a choice rather than a toggle, with options that progressively add in the 3 base cups, Hades cup, and then Ice Titan.

Did some test generations and looks to be working as intended. Presets were just based off vibes.

### CODE REVIEW

**Comment by esutley** [comment]
Could also see having Ice Titan be enabled by having Superbosses and Hades Cup both enabled rather than a specific setting for Cups. Don't know about including it in the final door conditions though

File: worlds/kh1/Options.py
Comment (gaithern) [inline worlds/kh1/Options.py:1]: We should maybe reword that first line to something like:
"Determines which cups have their locations added to the multiworld."

**Review by gaithern — CHANGES_REQUESTED** [review]

**Review by gaithern — APPROVED** [review]
